WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs

```diff
@@ -108,14 +106,17 @@ public async IAsyncEnumerable<CoinVerifyInfo> VerifyCoinsAsync(IEnumerable<Coin>
 		await foreach (var response in CoinVerifierApiClient.VerifyScriptsAsync(scriptsToCheck, linkedCts.Token))
 		{
 			bool shouldBanUtxo = CheckForFlags(response.ApiResponseItem);
-			var coin = coinDictionary[response.ScriptPubKey];
 
-			if (!shouldBanUtxo)
+			// Find all coins with the same script (address reuse). 
+			foreach (var coin in coinsToCheck.Where(c => c.ScriptPubKey == response.ScriptPubKey))
 			{
```

Comment:
Won't this take too long?

Reply:
Matter of milliseconds. If you want to optimize it you can do it in the next PR. You have to build a Dictionary<Script, Coin[]>.